src/sources/gcp_pubsub.rs

```diff
@@ -199,16 +199,21 @@ struct PubsubSource {
 
 impl PubsubSource {
     async fn run(mut self) -> crate::Result<()> {
+        while self.run_once().await? {}
```

Comment:
Under which conditions _would_ we like the source to exit on error?

Reply:
Well, it's not exiting on error but on shutdown. I just added a patch that moves all other errors of `run_once` except for connections, which cause an exit.

Reply:
Ah, but wouldn't the `?` there cause it to exit on error too?

Should we exit if there was a connection error? I would think we'd just retry.

Reply:
I see what you mean. Hmm, I think I'll punt that to a second issue and PR. This doesn't change that behavior, but AFAICT it _does_ fix the issue the OP had, as I saw it retry after that condition happened. We'll probably also want to add some configuration for that issue (retrying instantly is probably bad, etc).